### PR TITLE
chore(j-s): Implement tests for a search warrant on the court side

### PR DESCRIPTION
**Pull request by oddsson**
# Implement tests for a search warrant on the court side

[Asana](https://app.asana.com/0/1199153462262248/1206257892967291/f)

## What

Implement tests for a search warrant on the court side.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/system-e2e/src/tests/judicial-system/smoke/search-warrant-tests.spec.ts [hunk 1]
```diff
@@ -90,4 +91,59 @@ test.describe.serial('Search warrant tests', () => {
...
+        year: 'numeric',
+      })
+
+      await page.goto(`domur/rannsoknarheimild/mottaka/${caseId}`)
+      await page.getByTestId('courtCaseNumber').click()
+      await page.getByTestId('courtCaseNumber').fill(randomCourtCaseNumber())
+      await page
+        .getByText('Veldu dómara/aðstoðarmann *Veldu héraðsdómara')
+        .click()
+      await page.locator('#react-select-judge-option-1').click()
```
Comment (AndesKrrrrrrrrrrr) [inline apps/system-e2e/src/tests/judicial-system/smoke/search-warrant-tests.spec.ts:112]: Should use roles and names.

File: apps/system-e2e/src/tests/judicial-system/smoke/search-warrant-tests.spec.ts [hunk 2]
```diff
@@ -90,4 +91,59 @@ test.describe.serial('Search warrant tests', () => {
...
+      await page
+        .locator('label')
+        .filter({ hasText: 'Varnaraðili unir úrskurðinum' })
+        .click()
+      await page
+        .locator('label')
+        .filter({ hasText: 'Sækjandi tekur sér lögboðinn frest' })
+        .click()
+      await page.getByLabel('Dagsetning uppkvaðningar *').click()
+      await page.locator('input[id=courtEndTime]').fill(yesterday)
```
Comment (AndesKrrrrrrrrrrr) [inline apps/system-e2e/src/tests/judicial-system/smoke/search-warrant-tests.spec.ts:134]: Roles again. Not tested, only a suggestion
```suggestion
      await page.getByRole('date', {name: "Veldu lokadagsetningu"}).fill(yesterday)
```

File: apps/system-e2e/src/tests/judicial-system/smoke/search-warrant-tests.spec.ts [hunk 3]
```diff
@@ -90,4 +91,59 @@ test.describe.serial('Search warrant tests', () => {
...
+
+      await page.goto(`domur/rannsoknarheimild/mottaka/${caseId}`)
+      await page.getByTestId('courtCaseNumber').click()
+      await page.getByTestId('courtCaseNumber').fill(randomCourtCaseNumber())
+      await page
+        .getByText('Veldu dómara/aðstoðarmann *Veldu héraðsdómara')
+        .click()
+      await page.locator('#react-select-judge-option-1').click()
+      await page.getByTestId('continueButton').click()
+      await expect(page).toHaveURL(`domur/rannsoknarheimild/yfirlit/${caseId}`)
```
Comment (AndesKrrrrrrrrrrr) [inline apps/system-e2e/src/tests/judicial-system/smoke/search-warrant-tests.spec.ts:114]: Did you forget the root slash?
```suggestion
      await expect(page).toHaveURL(`/domur/rannsoknarheimild/yfirlit/${caseId}`)
```

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]
LGTM

**Review by unakb — APPROVED** [review]

File: apps/system-e2e/src/tests/judicial-system/smoke/search-warrant-tests.spec.ts (hunk 2 above)
Comment (oddsson) [reply apps/system-e2e/src/tests/judicial-system/smoke/search-warrant-tests.spec.ts:134]: This input (react-datepicker from island-ui) doesn't have a role date set. There are also two date pickers on this page, so I'm going to leave this as is for now.

**Pull request merged**